### PR TITLE
Added  yarn deploy:reset script to reset deployments

### LINKED ISSUE #209: Add script to reset deployments `yarn deploy:reset`
## Issue Overview

We need to allow users to keep previously deployed contracts that are stored locally after each deployment. 
The idea is to have the current `yarn deploy` to  not override the `latest` deployment but rather append contracts to it.

Add another script to do `yarn deploy:reset` which will do what `yarn:deploy` currently does. Reset all the contracts deployed that are exported to `packages/nextjs` to be only the contracts deployed on the last RUN.

## Proposed Solutions or Ideas

- should change the `package.json` on source folder and snfoundry package
- check that this works on devnet since devnet is usually started from zero, add an error message on missing contract declaration on UI. should tell you to run `yarn deploy:reset` if the contract inside `/debug` page doesnt exist

### PR DESCRIPTION
**Pull request by josephchimebuka**
Added the yarn deploy:reset script to reset deployments 
Fixes #209

### CODE REVIEW

File: packages/snfoundry/scripts-ts/deploy-reset.ts
Comment (0xquantum3labs) [inline packages/snfoundry/scripts-ts/deploy-reset.ts:1]: we shouldnt create another `deploy()` script that does the same as `deploy.ts`. we should only have a single `deploy.ts`, and the rest of the logic should be handled in `scripts-ts/deploy-contract.ts` we can use `yargs`, you can check how yargs is used on the file.

**Review by 0xquantum3labs — CHANGES_REQUESTED** [review]

File: packages/snfoundry/scripts-ts/deploy-reset.ts
Comment (josephchimebuka) [reply packages/snfoundry/scripts-ts/deploy-reset.ts:1]: Alright, I have made the changes

File: packages/snfoundry/scripts-ts/deploy.ts [hunk 1]
```diff
@@ -1,47 +1,30 @@
...
- *       }
- *     }
- *   );
- * };
- *
- *
- * @returns {Promise<void>}
- */
 const deployScript = async (): Promise<void> => {
+  if (argv.reset) {
```
Comment (0xquantum3labs) [inline packages/snfoundry/scripts-ts/deploy.ts:19]: can we leave the `deployScript` as before, and use this args inside `deploy-contract.ts` only ?

**Review by 0xquantum3labs — CHANGES_REQUESTED** [review]

**Comment by Nadai2010** [comment]
Hello, @josephchimebuka we remind you that #ODHack ends tomorrow, please review comments and resolve as soon as possible.

File: packages/snfoundry/scripts-ts/deploy.ts (hunk 1 above)
Comment (josephchimebuka) [reply packages/snfoundry/scripts-ts/deploy.ts:19]: Yes we can... Sending a PR now

Comment (0xquantum3labs) [reply packages/snfoundry/scripts-ts/deploy.ts]: i still see the `deploy.ts` file same as before, we should be able to put all the logic inside `deploy-contract.ts` @josephchimebuka

Comment (josephchimebuka) [reply packages/snfoundry/scripts-ts/deploy.ts]: I made some new updates and put the logic into deploy- contract.ts files

Comment (jrcarlos2000) [reply packages/snfoundry/scripts-ts/deploy.ts]: Hello @josephchimebuka , thanks for this. The file 'deploy.ts' should remain as the main branch. ( including the comments ) 

All the logic goes into 'deploy-contract.ts'

Comment (josephchimebuka) [reply packages/snfoundry/scripts-ts/deploy.ts]: @0xquantum3labs I've returned the deploy.ts to the default one

File: packages/snfoundry/scripts-ts/deploy-contract.ts [hunk 2]
```diff
@@ -230,28 +254,61 @@ const executeDeployCalls = async (options?: UniversalDetails) => {
...
+  const networkPath = path.resolve(
+    __dirname,
+    `../deployments/${networkName}_latest.json`
+  );
+  if (fs.existsSync(networkPath)) {
+    return JSON.parse(fs.readFileSync(networkPath, "utf8"));
+  }
+  return {};
+};
+// const exportDeployments = () => {
```
Comment (0xquantum3labs) [inline packages/snfoundry/scripts-ts/deploy-contract.ts:267]: lets remove the lines that are commented out @josephchimebuka

File: packages/snfoundry/scripts-ts/deploy-contract.ts [hunk 3]
```diff
@@ -14,8 +14,32 @@ import {
 import { DeployContractParams, Network } from "./types";
 import { green, red, yellow } from "./helpers/colorize-log";
 
-const argv = yargs(process.argv.slice(2)).argv;
-const networkName: string = argv["network"];
+// const argv = yargs(process.argv.slice(2)).argv;
```
Comment (0xquantum3labs) [inline packages/snfoundry/scripts-ts/deploy-contract.ts:17]: lets remove the lines that are commented out @josephchimebuka

**Review by 0xquantum3labs — CHANGES_REQUESTED** [review]

**Review by 0xquantum3labs — CHANGES_REQUESTED** [review]
lets delete the lines that are commented out , the rest seems good to me @josephchimebuka

**Comment by josephchimebuka** [comment]
I have deleted the commented lines out @0xquantum3labs

File: packages/snfoundry/scripts-ts/deploy-contract.ts [hunk 4]
```diff
@@ -65,26 +87,7 @@ const deployContract_NotWait = async (payload: {
   }
 };
 
-/**
- * Deploy a contract using the specified parameters.
```
Comment (0xquantum3labs) [inline packages/snfoundry/scripts-ts/deploy-contract.ts:69]: only the comments you added, not the previous ones, these ones serve as explanation

Comment (0xquantum3labs) [reply packages/snfoundry/scripts-ts/deploy-contract.ts:69]: @josephchimebuka

**Review by 0xquantum3labs — CHANGES_REQUESTED** [review]

File: packages/snfoundry/scripts-ts/deploy-contract.ts (hunk 4 above)
Comment (josephchimebuka) [reply packages/snfoundry/scripts-ts/deploy-contract.ts:69]: oh okay i'll return it now

**Review by 0xquantum3labs — APPROVED** [review]

**Review by Nadai2010 — APPROVED** [review]

**Pull request merged**